tests/e2e/actions.go

```diff
@@ -268,72 +391,220 @@ type SubmitConsumerAdditionProposalAction struct {
-		params.HistoricalEntries,
-		action.DistributionChannel,
-		action.TopN,
-		action.ValidatorsPowerCap,
-		action.ValidatorSetCap,
-		action.Allowlist,
-		action.Denylist,
-		action.AllowInactiveVals,
-		action.MinStake,
-		action.Deposit)
+	spawnTime := tr.testConfig.containerConfig.Now.Add(time.Duration(action.SpawnTime) * time.Millisecond)
 
-	//#nosec G204 -- bypass unsafe quoting warning (no production code)
+	Metadata := types.ConsumerMetadata{
+		Name:        "chain " + string(action.Chain),
+		Description: "no description",
+		Metadata:    "no metadata",
+	}
+
+	InitializationParameters := types.ConsumerInitializationParameters{
+		InitialHeight: action.InitialHeight,
+		GenesisHash:   []byte("gen_hash"),
+		BinaryHash:    []byte("bin_hash"),
+		SpawnTime:     spawnTime,
+
+		UnbondingPeriod:                   params.UnbondingPeriod,
+		CcvTimeoutPeriod:                  params.CcvTimeoutPeriod,
+		TransferTimeoutPeriod:             params.TransferTimeoutPeriod,
+		ConsumerRedistributionFraction:    params.ConsumerRedistributionFraction,
+		BlocksPerDistributionTransmission: params.BlocksPerDistributionTransmission,
+		HistoricalEntries:                 params.HistoricalEntries,
+		DistributionTransmissionChannel:   action.DistributionChannel,
+	}
+
+	consumerId := tr.CreateConsumer(action.Chain, action.ConsumerChain, action.From, Metadata, nil, nil)
+	authority := "cosmos10d07y265gmmuvt4z0w9aw880jnsr700j6zn9kn"
+
+	// Update consumer to change owner to governance before submitting the proposal
+	update := &types.MsgUpdateConsumer{
+		ConsumerId:      string(consumerId),
+		NewOwnerAddress: authority,
+	}
+	// For the MsgUpdateConsumer sent in the proposal
+	PowerShapingParameters := types.PowerShapingParameters{
+		Top_N:              0,
+		ValidatorsPowerCap: action.ValidatorsPowerCap,
+		ValidatorSetCap:    action.ValidatorSetCap,
+		Allowlist:          action.Allowlist,
+		Denylist:           action.Denylist,
+		MinStake:           action.MinStake,
+		AllowInactiveVals:  action.AllowInactiveVals,
+	}
+	update.PowerShapingParameters = &PowerShapingParameters
+	tr.UpdateConsumer(action.Chain, action.From, *update)
+
+	// - set PowerShaping params TopN > 0 for consumer chain
+	update.PowerShapingParameters.Top_N = action.TopN
+	update.Signer = authority
+	update.InitializationParameters = &InitializationParameters
+	update.InitializationParameters.SpawnTime = spawnTime
+	update.Metadata = &Metadata
+
+	// Generate proposal content
+	title := "Propose the addition of a new chain"
+	description := "description of the consumer modification proposal"
+	summary := "Gonna be a great chain"
+	expedited := false
+	metadata := "ipfs://CID"
+	deposit := fmt.Sprintf("%dstake", action.Deposit)
+	jsonStr := e2e.GenerateGovProposalContent(title, summary, metadata, deposit, description, expedited, update)
+
+	// #nosec G204 -- bypass unsafe quoting warning (no production code)
 	proposalFile := "/consumer-addition.proposal"
```

Comment:
Should this file have a `.json` suffix for clairty.
Also, can we consistently use `_` or `-` (e.g., `jsonFile := "/create_consumer.json"` versus `proposalFile := "/consumer-addition.proposal"`)?